Remotion/Core/Core/Reflection/TypeExtensions.cs

```diff
@@ -189,25 +150,17 @@ private static IReadOnlyList<Type> GetAscribedGenericClassArgumentsInternal (Typ
         throw ArgumentUtility.CreateArgumentTypeException ("type", type, ascribeeType);
     }
 
-    private static bool CanDirectlyAscribeToGenericTypeInternalFromCache (Type type, Type ascribeeType, Type ascribeeGenericTypeDefinition)
-    {
-      // C# compiler 7.2 already provides caching for anonymous method.
-      return s_canDirectlyAscribeToGenericTypeInternalCache.GetOrAdd (
-        Tuple.Create (type, ascribeeType, ascribeeGenericTypeDefinition),
-        key => CanDirectlyAscribeToGenericTypeInternal (key.Item1, key.Item2, key.Item3));
-    }
-
     private static bool CanDirectlyAscribeToGenericTypeInternal (Type type, Type ascribeeType, Type ascribeeGenericTypeDefinition)
     {
       Assertion.IsNotNull (ascribeeType);
 
-      if (!type.IsGenericType || type.GetGenericTypeDefinition () != ascribeeGenericTypeDefinition)
+      if (!type.IsGenericType || type.GetGenericTypeDefinition() != ascribeeGenericTypeDefinition)
         return false;
 
       if (ascribeeType != ascribeeGenericTypeDefinition)
         return ascribeeType.IsAssignableFrom (type);
       else
-        return ascribeeType.IsAssignableFrom (type.GetGenericTypeDefinition ());
+        return ascribeeType.IsAssignableFrom (type.GetGenericTypeDefinition());
     }
   }
-}
+}
```

Comment:
ReflectionUtility.GetObjectListTypeParameter
probably cache. Maybe a single cache with IsObjectList result since we will need the typeparameter anyway for any ObjectList<> we find.